File: problems/leetcode/find-minimum-in-rotated-sorted-array.java
// https://leetcode.com/problems/find-minimum-in-rotated-sorted-array/
class Solution {
    public int findMin(int[] nums) {
        int left = 0, right = nums.length - 1;
        int rightVal = nums[right];
        
        while (left < right) {
            int mid = (left + right) / 2;
            if (nums[mid] < rightVal) {
                right = mid;
            } else {
                left = mid + 1;
            }
        }
        
        return nums[left];
    }
}

File: problems/leetcode/find-minimum-in-rotated-sorted-array.py
# https://leetcode.com/problems/find-minimum-in-rotated-sorted-array/
class Solution:
    def findMin(self, nums: List[int]) -> int:
        def findOffset() -> int:
            start, end = 0, len(nums) - 1
            while start < end:
                mid = (start + end) // 2
                if nums[mid] > nums[end]:
                    start = mid + 1
                else:
                    end = mid

            return start

        return nums[findOffset()]
